cpp/src/DataStorm/TopicI.h

```diff
@@ -28,7 +28,7 @@ namespace DataStormI
     public:
         TopicI(
             std::shared_ptr<Instance>,
-            std::shared_ptr<TopicFactoryI>,
+            const std::shared_ptr<TopicFactoryI>&,
```

Comment:
Here, switching to `const shared_ptr&` makes sense: we don't adopt a refcount but create and keep a weak_ptr from this `const shared_ptr&`.